recipes/google-cloud-cpp-core/meta.yaml

```diff
@@ -0,0 +1,364 @@
+    - {{ pin_subpackage("libgoogle-cloud-common", max_pin="x.x") }}
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+  host:
+    - {{ pin_subpackage("libgoogle-cloud-common", exact=True) }}
+  run:
+    - {{ pin_subpackage("libgoogle-cloud-common", exact=True) }}
+    - libabseil
+    - libprotobuf
+    - libgrpc
+    - libcxx  # [osx]
+test:
+  {{ test_subpackage(feature)|indent(2) }}
+{%- endmacro %}
+
+{%- macro standard_devel_subpackage(feature) %}
+script: build-libgoogle-cloud.sh   # [not win]
+script: build-libgoogle-cloud.bat  # [win]
+build:
+  run_exports:
+    # patch versions guaranteed to be ABI-compatible, see
+    # https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2023-05-03-patch-releases-are-ABI-compatible.md
+    - {{ pin_subpackage("libgoogle-cloud-common-devel", max_pin="x.x") }}
+    - {{ pin_subpackage("libgoogle-cloud-" + feature, max_pin="x.x") }}
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+  host:
+    - {{ pin_subpackage("libgoogle-cloud-common-devel", exact=True) }}
+  run:
+    - {{ pin_subpackage("libgoogle-cloud-common-devel", exact=True) }}
+    - {{ pin_subpackage("libgoogle-cloud-" + feature, exact=True) }}
+    - libcxx  # [osx]
+test:
+  {{ test_devel_subpackage(feature)|indent(2) }}
+{% endmacro %}
+
+{%- macro child_subpackage(feature, parent) %}
+script: build-libgoogle-cloud.sh   # [not win]
+script: build-libgoogle-cloud.bat  # [win]
+build:
+  run_exports:
+    # patch versions guaranteed to be ABI-compatible, see
+    # https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2023-05-03-patch-releases-are-ABI-compatible.md
+    - {{ pin_subpackage("libgoogle-cloud-" + parent, max_pin="x.x") }}
+    - {{ pin_subpackage("libgoogle-cloud-common", max_pin="x.x") }}
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - cmake
+    - ninja
+  host:
+    - {{ pin_subpackage("libgoogle-cloud-" + parent, exact=True) }}
+    - {{ pin_subpackage("libgoogle-cloud-common", exact=True) }}
+  run:
+    - {{ pin_subpackage("libgoogle-cloud-" + parent, exact=True) }}
+    - {{ pin_subpackage("libgoogle-cloud-common", exact=True) }}
+    - libabseil
+    - libprotobuf
+    - libgrpc
+    - libcxx  # [osx]
+test:
+  {{ test_subpackage(feature)|indent(2) }}
+{%- endmacro %}
+
+{%- macro child_devel_subpackage(feature, parent) %}
+script: build-libgoogle-cloud.sh   # [not win]
+script: build-libgoogle-cloud.bat  # [win]
+build:
+  run_exports:
+    - {{ pin_subpackage("libgoogle-cloud-" + feature, max_pin="x.x") }}
+    # patch versions guaranteed to be ABI-compatible, see
+    # https://github.com/googleapis/google-cloud-cpp/blob/main/doc/adr/2023-05-03-patch-releases-are-ABI-compatible.md
+    - {{ pin_subpackage("libgoogle-cloud-" + parent + "-devel", max_pin="x.x") }}
+    - {{ pin_subpackage("libgoogle-cloud-common-devel", max_pin="x.x") }}
```

Comment:
We shouldn't run-export `devel` to `devel`. The run-exports should generally go `libfoo-devel` -> `libfoo` (possibly also `libbar` and `libbaz` if necessary). Artefacts built on top do not need the headers of the metadata to run, only the correct libraries.

Reply:
Thanks, this helps.  Do you have a good recommendation on where I could learn about how to use `build/run_exports` vs. `build` vs. `host` vs. `run` dependencies?  In particular, how they are applied transitively? 

 I fear I am doing things wrong because I don't understand what these are trying to describe, or that the difference between a `run` vs. `host` dependency is too subtle for me (I am sure it is not subtle, I am just not understanding something).